### ISSUE TITLE
Can't create series from `list[numpy | None]`

### ISSUE THREAD

**Issue by s-banach** [pola-rs/polars#16633]
### Checks

- [X] I have checked that this issue has not already been reported.
- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Reproducible example

```python
import numpy as np
import polars as pl

t = pl.Series([np.array([1, 2]), None, np.array([3, 4])])
```

### Issue description

The following code is from `sequence_to_pyseries`:

```python
    elif (
        _check_for_numpy(value)
        and isinstance(value, np.ndarray)
        and len(value.shape) == 1
    ):
        n_elems = len(value)
        if all(len(v) == n_elems for v in values):
            # can take (much) faster path if all lists are the same length
            return numpy_to_pyseries(
                name,
                np.vstack(values),
                strict=strict,
                nan_to_null=nan_to_null,
            )
        else:
            return PySeries.new_series_list(
                name,
                [
                    numpy_to_pyseries("", v, strict=strict, nan_to_null=nan_to_null)
                    for v in values
                ],
                strict,
            )
```

I guess it would fix the issue to modify the code like this:

```python
    ...
        if all(isinstance(v, np.ndarray) and len(v) == n_elems for v in values):
            ...
        else:
            ...
                    None if v is None else numpy_to_pyseries("", v, strict=strict, nan_to_null=nan_to_null)
                    for v in values
            ...
```

But I'm not sure if there's anything I'm missing or if there's a better way.

### Expected behavior

Allow mix of Null and numpy array.

### Installed versions

<details>

```
--------Version info---------
Polars:               0.20.30
Index type:           UInt32
Platform:             Windows-10-10.0.19045-SP0
Python:               3.12.2 | packaged by conda-forge | (main, Feb 16 2024, 20:42:31) [MSC v.1937 64 bit (AMD64)]

----Optional dependencies----
adbc_driver_manager:  <not installed>
cloudpickle:          <not installed>
connectorx:           <not installed>
deltalake:            <not installed>
fastexcel:            0.10.4
fsspec:               2024.3.1
gevent:               <not installed>
hvplot:               <not installed>
matplotlib:           3.8.4
nest_asyncio:         1.6.0
numpy:                1.26.4
openpyxl:             <not installed>
pandas:               2.2.2
pyarrow:              16.0.0
pydantic:             2.7.2
pyiceberg:            <not installed>
pyxlsb:               <not installed>
sqlalchemy:           2.0.30
torch:                2.3.0+cpu
xlsx2csv:             <not installed>
xlsxwriter:           <not installed>
```

</details>

**Comment by glthm** [comment]
Hi, to give more insight on this issue:

With lists and `None` (this is the **actual output**):
```python
>>> pl.Series([[1,2], None, [3,4]])
shape: (3,)
Series: '' [list[i64]]
[
    [1, 2]
    null
    [3, 4]
]
```

With numpy arrays and `None` (this is the **expected output** of the code given by OP):
```python
>>> pl.Series([np.array([1,2]), None, np.array([3,4])])
shape: (3,)
Series: '' [array[i64, 3]]
[
    [1, 2]
    null
    [3, 4]
]
```

This is the **actual output**:
```python
>>> pl.Series([np.array([1,2]), None, np.array([3,4])])
TypeError: object of type 'NoneType' has no len()
```

Same with integers, floats, etc. instead of `None` (**actual output**):
```python
>>> pl.Series([np.array([1,2]), 5, np.array([3,4])], strict=False)
TypeError: object of type 'int' has no len()
```

For lists, this is possible (only with the `strict=False` argument), even if it's not the cleanest thing to do in your code of course (**actual output**):
```python
>>> pl.Series([np.array([1,2]), 5, np.array([3,4])], strict=False)
shape: (3,)
Series: '' [o][object]
[
     [1, 2]
     5
     [3, 4]
]
```

Therefore, the **expected output** for objects having no length would be something looking like:
```python
>>> pl.Series([np.array([1,2]), 5, np.array([3,4])], strict=False)
shape: (3,)
Series: '' [o][object]
[
     np.array([1, 2], dtype=i64)
     5
     np.array([3, 4], dtype=i64)
]
```

Note that, if the first item given is not a numpy array, this is the **actual output** for integers (not for `None` values however):
```python
>>> pl.Series([5, np.array([1,2]), np.array([3,4])], strict=False)
shape: (3,)
Series: '' [o][object]
[
     5
     [1, 2]
     [3, 4]
]
>>> pl.Series([None, np.array([1,2]), np.array([3,4])], strict=False)
TypeError: object of type 'NoneType' has no len()
```

**Comment by glthm** [comment]
Trying to look further into this

So in that case, [we are in this branch of the code](https://github.com/pola-rs/polars/blob/11dd4b3e920e8c3bf151fa56e00b686326cd2410/py-polars/polars/_utils/construction/series.py#L229), shown here by OP

> ```python
>     elif (
>         _check_for_numpy(value)
>         and isinstance(value, np.ndarray)
>         and len(value.shape) == 1
>     ):
>         n_elems = len(value)
>         if all(len(v) == n_elems for v in values):
>             # can take (much) faster path if all lists are the same length
>             return numpy_to_pyseries(
>                 name,
>                 np.vstack(values),
>                 strict=strict,
>                 nan_to_null=nan_to_null,
>             )
>         else:
>             return PySeries.new_series_list(
>                 name,
>                 [
>                     numpy_to_pyseries("", v, strict=strict, nan_to_null=nan_to_null)
>                     for v in values
>                 ],
>                 strict,
>             )
> ```

`values` is the `Sequence[Any]` to put in our `pl.Series`, and `value` is the first non-null value of `values`, which is a 1D numpy array if we are in this `elif` branch. However, indeed [L235](https://github.com/pola-rs/polars/blob/11dd4b3e920e8c3bf151fa56e00b686326cd2410/py-polars/polars/_utils/construction/series.py#L235) breaks if some of the `v` in `values` have no `__len__` attribute, and this is not detected by linting as `values` is a `Sequence[Any]` in the function declaration (as it should be).

Proposed solution, rather than only catching  the `None` case, I'd keep the beginning of the proposed answer but not alter the treatment of numpy arrrays that have different sizes, and simply delegate the case of mixed arrays and not-arrays to the final `else` condition of the `sequence_to_pyseries` function body, that will ultimately handle it as it handles the case of lists, with `PySeries.new_from_any_values(...)` in [L292](https://github.com/pola-rs/polars/blob/11dd4b3e920e8c3bf151fa56e00b686326cd2410/py-polars/polars/_utils/construction/series.py#L292). This also avoids any unforeseen pitfall (hopefully!).

Therefore, a fix could be to only edit the clause to enter the `elif` branch:
```python
     elif (
         _check_for_numpy(value)
         and isinstance(value, np.ndarray)
         and len(value.shape) == 1
         # vvvvvvvvvvvvvvvvv here vvvvvvvvvvvvvvvvvvvvv
         and all(isinstance(v, np.ndarray) for v in values)
         # ^^^^^^^^^^^^^^^^^ here ^^^^^^^^^^^^^^^^^^^^
     ):
         n_elems = len(value)
         if all(len(v) == n_elems for v in values):
             # can take (much) faster path if all lists are the same length
             return numpy_to_pyseries(
                 name,
                 np.vstack(values),
                 strict=strict,
                 nan_to_null=nan_to_null,
             )
         else:
             return PySeries.new_series_list(
                 name,
                 [
                     numpy_to_pyseries("", v, strict=strict, nan_to_null=nan_to_null)
                     for v in values
                 ],
                 strict,
             )
```


I do not have the means to test it much more but I hope it helped a bit to solve the issue. If nobody objects to it I'll try to go through the process of testing this and proposing a merge request in the next days/weeks

**Comment by glthm** [comment]
### TL;DR

- Issue should be renamed Can't create series from Sequence[numpy | Any]

- I proposed a fix above
